src/transaction_builder.cpp

```diff
@@ -311,3 +424,285 @@ TransactionBuilderResult TransactionBuilder::Build()
 
     return TransactionBuilderResult(CTransaction(mtx));
 }
+
+void TransactionBuilder::CreateJSDescriptions()
+{
+    // Copy jsInputs and jsOutputs to more flexible containers
+    std::deque<libzcash::JSInput> jsInputsDeque;
+    for (auto jsInput : jsInputs) {
+        jsInputsDeque.push_back(jsInput);
+    }
+    std::deque<libzcash::JSOutput> jsOutputsDeque;
+    for (auto jsOutput : jsOutputs) {
+        jsOutputsDeque.push_back(jsOutput);
+    }
+
+    // If we have no Sprout shielded inputs, then we do the simpler more-leaky
+    // process where we just create outputs directly. We save the chaining logic,
+    // at the expense of leaking the sums of pairs of output values in vpub_old.
```

Comment:
This is a partial regression of the problem documented in #1360 and fixed in #2440. If we were to switch to using this code rather than the existing builder, these transactions would also be distinguishable from those created by the existing builder, which is a client version distinguisher. That should be documented explicitly.

Reply:
@str4d, do you have thoughts on this?

Reply:
I think I misread the code. This algorithm is only used in the case where we have no Sprout inputs, so there is no information to leak about the Sprout input note values.